test_util/azure_test_driver.py

```diff
@@ -124,14 +120,17 @@ def main():
         print("Creating template deployment ...")
         deploy_poller = rmc.deployments.create_or_update(group_name, deployment_name, deployment_properties)
 
-        # Stop after 45 attempts (each one takes up to one minute)
-        @retry(stop_max_attempt_number=45)
+        def retry_if_assertion_error(exception):
+            """Return True if we should retry (in this case when it's an AssertionError), False otherwise"""
+            return isinstance(exception, AssertionError)
+
+        @retry(retry_on_exception=retry_if_assertion_error, stop_max_attempt_number=45)
```

Comment:
Actually, retrying on AssertionError only is the right thing to do. The way the code is written, AssertionError is part of the logic for checking if the task is done and a retry is a while loop to for waiting until the task is complete. 

The unfortunate scenario when CloudError on Azure happens and it kills the task, we should not be retrying for 45 times, the clouderror must have killed the long running task and we should bail out then and exit with a failure.  This our fix (fail fast) is the fix the flaky azure tests which are not in our control.

HTH.